infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/server/IServerChannelAdapter.java

```diff
@@ -42,12 +38,28 @@ public IServerChannelAdapter(ServerContext serverContext, LogReplicationServerRo
     public abstract void send(CorfuMessage.ResponseMsg msg);
 
     /**
-     * Receive a message from Client.
+     * Send a message across the channel to a specific endpoint.
+     *
+     * @param request corfu message to be sent
+     */
+    public abstract void send(CorfuMessage.RequestMsg request);
```

Comment:
why do we not required a node id here but it is required in the ClientChannelAdapter?

Reply:
In ClientChannelAdapter, nodeId is required to find the channel to send the request.
But on the server side, we have something called a "streamObserver". Its a GRPC construct. 
Basically there is 1 streamObserver per rpc. And on server, we just find the streamObserver for the request+session received, and use that to send msgs from the server.